docs/VirtualMachine.md

```diff
@@ -0,0 +1,160 @@
+# Virtual Machine Design
+
+This document details the workings and design of the virtual machine that executes a Flux script.
+
+## Execution Stack
+
+A Flux script goes through several transformations from text to execution.
+
+1. Flux source code is created via any means.
+2. The parser produces an abstract syntax tree (AST) from the source code.
+3. The semantic analyzer produces a semantic graph from the AST.
+4. The compiler accepts the semantic graph and produces an intermediate representation (IR) to be executed.
+5. The flux virtual machine executes the IR.
+
+NOTE: Should we call it bytecode or IR?
+Should we call it a VM or an interpreter?
```

Comment:
I'd call it a VM, as the source code must be compiled to some intermediate representation / byte code before it can be executed. Conceivably, I could create DSL "foo" that compiles to Flux IR and the VM would execute my code.